x-pack/plugins/observability/public/pages/cases/use_fetch_alert_detail.test.ts

```diff
@@ -0,0 +1,161 @@
+        } as unknown as PluginContextValue)
+    );
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('initially is not loading and does not have data', async () => {
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<string, [boolean, TopAlert | null]>(() =>
+        useFetchAlertDetail(id)
+      );
+
+      await waitForNextUpdate();
+
+      expect(result.current).toEqual([false, null]);
+    });
+  });
+
+  it('returns no data when an error occurs', async () => {
+    mockUseKibanaReturnValue.services.http.get.mockImplementation(async () => {
+      throw new Error('an http error');
+    });
+
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<string, [boolean, TopAlert | null]>(() =>
+        useFetchAlertDetail('123')
+      );
+
+      await waitForNextUpdate();
+
+      expect(result.current).toEqual([false, null]);
+    });
+  });
+
+  it('retrieves the alert data', async () => {
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<string, [boolean, TopAlert | null]>(() =>
+        useFetchAlertDetail(id)
+      );
+
+      await waitForNextUpdate();
+      await waitForNextUpdate();
+
+      expect(result.current).toMatchInlineSnapshot(`
+        Array [
+          false,
+          Object {
+            "0": "a",
+            "1": " ",
+            "2": "r",
+            "3": "e",
+            "4": "a",
+            "5": "s",
+            "6": "o",
+            "7": "n",
+            "active": true,
+            "fields": Object {
+              "@timestamp": "2022-01-31T18:20:57.204Z",
+              "event.action": "active",
+              "event.kind": "signal",
+              "kibana.alert.duration.us": 13793555000,
+              "kibana.alert.instance.id": "*",
+              "kibana.alert.reason": "Document count has reported no data over the past 1h for *",
+              "kibana.alert.rule.category": "Metric threshold",
+              "kibana.alert.rule.consumer": "infrastructure",
+              "kibana.alert.rule.execution.uuid": "e62c418d-734d-47e7-bbeb-e6f182f5fb45",
+              "kibana.alert.rule.name": "A super rule",
+              "kibana.alert.rule.producer": "infrastructure",
+              "kibana.alert.rule.rule_type_id": "metrics.alert.threshold",
+              "kibana.alert.rule.tags": Array [],
+              "kibana.alert.rule.uuid": "69411af0-82a2-11ec-8139-c1568734434e",
+              "kibana.alert.start": "2022-01-31T14:31:03.649Z",
+              "kibana.alert.status": "active",
+              "kibana.alert.uuid": "73c0d0cd-2df4-4550-862c-1d447e9c1db2",
+              "kibana.alert.workflow_status": "open",
+              "kibana.space_ids": Array [
+                "default",
+              ],
+              "kibana.version": "8.1.0",
+              "tags": Array [],
+            },
+            "link": undefined,
+            "reason": "A super rule",
```

Comment:
@jonathan-buttner I've never seen such a field in the alerts-as-data indices? What is it? This is outside of the `fields`, right?

Reply:
I'm found this mock in another test I think: https://github.com/elastic/kibana/pull/123883/files/015b351f1aa3ac7911bf89441ef09d6aa8f0ced1#diff-dc8f3da954e34ac852d88a16d665fe0455fa59ff41561c7fc6a316e4c4e9a3e6R49